### PR TITLE
use clang-tidy to check  c10/util/*cpp

### PR DESCRIPTION
**Pull request by cyyever**
This PR increases the coverage of clang-tidy to c10/util/*cpp and fixes some errors.

### CODE REVIEW

**Comment by cyyever** [comment]
@Skylion007 We can use clang-tidy in c10/cuda because the docker image for lintrunner has no CUDA packages.

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
@cyyever The C10 header filter regex also needs to be updated: https://github.com/pytorch/pytorch/blob/3864207c2a71a3ba8dc13bcf9582a726a10292cd/.clang-tidy#LL49C1-L49C82 Note that clang-tidy doesn't support negative lookaheads so the (?!test) part actually probably doesn't work properly.

File: c10/util/int128.cpp [hunk 1]
```diff
@@ -57,13 +57,14 @@ const uint128_pod kuint128max = {
   } while (0)
 static inline int Fls64(uint64_t n) {
   //   GOOGLE_DCHECK_NE(0, n);
-  int pos = 0;
```
Comment (Skylion007) [inline c10/util/int128.cpp:60]: shouldn't at least be int64 or am I missing something?

**Comment by Skylion007** [comment]
@cyyever The next big task will be incrementally enabling this in `aten/src` . Maybe best to start with `aten/src/core` ?

**Pull request closed without merging**